docs/enhancements/312-add-flavorAssignmentStrategy-to-localQueue/README.md

```diff
@@ -0,0 +1,817 @@
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+To better representing the priorities of different flavors under each resource, we'd like to add a new field `priority` for each flavor.
+
+What's more, when finding satisfied flavors in scheduling, we'll iterate the flavors defined in a golang slice struct in sequence to
+find the satisfied ones, this is too primitive so we'd like to support more assigning strategies to meet more scenarios.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Currently, for flavors under one resource, the indexes of these flavors implicitly represents the flavor priorities. This is not descriptive enough,
+like we want to add two flavors with the same priority, this is helpful when we want to apply more assigning policies, like minimum borrowing from
+shared clusterQueues with same priority flavors. So we'd like to add a new field `Priority` to `Flavor`.
+
+As we described in the above section, when finding satisfied flavors, we follow the semantic in-order strategy by iterating the flavors, and we'll
+borrow resources whenever we're insufficient of specified resources. This is a simple way but can't meet various scenarios. E.g. we want to borrow
+minimum resources from other co-sharing clusterQueues to mitigate the disruptions once preemption is supported. Another case is when borrowing, we'd
+like to keep the resource balance within each clusterQueue, or the corresponding pods will stuck in pending for exhausted of several special resources.
+
+To achieve this, we'll introduce a new field `FlavorAssignmentStrategy` to `LocalQueue`, and we'll implement multiple strategies.
```

Comment:
Why `FlavorAssignmentStrategy` should be placed in `LocalQueue` instead of `ClusterQueue`?